lib/datadog/appsec/contrib/rack/gateway/response.rb

```diff
@@ -19,9 +19,58 @@ def initialize(body, status, headers, scope:)
               @scope = scope
             end
 
+            def parsed_body
+              return unless Datadog.configuration.appsec.parse_response_body
+
+              unless body.instance_of?(Array) || body.instance_of?(::Rack::BodyProxy)
+                if result.timeout
+                  Datadog.logger.debug do
+                    "Unable to parse response body because of unsupported body type: #{body.class}"
+                  end
+                end
+                return
+              end
+              return unless supported_response_type
+
+              body_dup = body.dup # avoid interating over the body. This is just in case code.
+              result = ''.dup
+              all_body_parts_are_string = true
+
+              body_dup.each do |body_part|
+                if body_part.is_a?(String)
+                  result.concat(body_part)
+                else
+                  all_body_parts_are_string = false
+                  break
+                end
+              end
+
+              return unless all_body_parts_are_string
+
+              if json?
+                JSON.parse(result)
+              else
+                result
```

Comment:
Then it's not really parsed, is it? Either it's raw (`server.response.body.raw`) or it's parsed (`server.response.body`).

Since the goal is to parse it - notably to perform schema extraction - it seems to me there's not much point doing that.